src/adapt.jl

```diff
@@ -65,39 +52,46 @@ function do_quadgk(f, s, n, ::Type{Tw}, atol, rtol, maxevals, nrm) where Tw
     return (I, E)
 end
 
-# handle keyword deprecation
-function tols(atol,rtol,abstol,reltol)
-    if !ismissing(abstol) || !ismissing(reltol)
-        Base.depwarn("abstol and reltol keywords are now atol and rtol, respectively", :quadgk)
+function handle_infinities(f, s, n, atol, rtol, maxevals, nrm)
+    if eltype(s) <: Real # check for infinite or semi-infinite intervals
+        s1 = s[1]; s2 = s[end]; inf1 = isinf(s1); inf2 = isinf(s2)
+        if inf1 || inf2
+            if inf1 && inf2 # x = t/(1-t^2) coordinate transformation
+                return do_quadgk(t -> begin t2 = t*t; den = 1 / (1 - t2);
+                                            f(t*den) * (1+t2)*den*den; end,
+                                map(x -> isinf(x) ? copysign(one(x), x) : 2x / (1+hypot(1,2x)), s),
+                                n, atol, rtol, maxevals, nrm)
+            end
+            let (s0,si) = inf1 ? (s2,s1) : (s1,s2)
```

Comment:
Why is the `let` necessary?

Reply:
JuliaLang/julia#15276 … since `s0` is used as a captured variable, it gets inferred as `Any` without the `let`.

I'll add a comment.

Reply:
Ah okay, thanks.